src/main/java/org/example/security/CustomUserDetailService.java

```
@@ -0,0 +1,7 @@
+package org.example.security;
+
+import org.springframework.security.core.userdetails.UserDetails;
+
+public interface CustomUserDetailService {
+    UserDetails loadUserByUsername(String email);
```

The method 'loadUserByUsername' should throw a UsernameNotFoundException if the user is not found, as per the contract of the UserDetailsService interface in Spring Security. Consider updating the method signature to include 'throws UsernameNotFoundException'.